src/tools/tidy/src/x_version.rs

```diff
@@ -0,0 +1,36 @@
+use semver::{BuildMetadata, Prerelease, Version};
+use std::io::ErrorKind;
+use std::process::{Command, Stdio};
+
+pub fn check(bad: &mut bool) {
+    let result = Command::new("x").arg("--version").stdout(Stdio::piped()).spawn();
+    let child = match result {
+        Ok(child) => child,
+        Err(e) => match e.kind() {
+            ErrorKind::NotFound => return,
+            _ => return tidy_error!(bad, "{}", e),
+        },
+    };
+
+    let output = child.wait_with_output().unwrap();
+
+    if output.status.success() {
+        let version = String::from_utf8_lossy(&output.stdout);
+        let version = Version::parse(version.trim_end()).unwrap();
```

Comment:
This still has the problem I mentioned in https://github.com/rust-lang/rust/pull/104552#discussion_r1026901085.
```
; x test tidy
tidy error: exit status: 1
```
(as an aside, the error message here is unhelpful - left a suggestion in another comment)